mumd/src/network/udp.rs

```diff
@@ -190,6 +194,15 @@ async fn send_ping(sink: &mut UdpSender, server_addr: SocketAddr) {
         .unwrap();
 }
 
+async fn send_pings(sink: Arc<Mutex<UdpSender>>, server_addr: SocketAddr) {
```

Comment:
We'd probably want to check if we get a response to the pings and if not, start tunneling audio through TCP.

Reply:
We already do, actually; we start by always sending audio over TCP:
https://github.com/sornas/mum/blob/2264ce4eb7922d9c7313d64f85649e861207842b/mumd/src/state.rs#L582-L585
and switch to UDP when we receive a ping response:
https://github.com/sornas/mum/blob/2264ce4eb7922d9c7313d64f85649e861207842b/mumd/src/network/udp.rs#L152-L157

Reply:
When we stop receiving responses to our UDP ping, we probably want to tunnel our audio through TCP, since probably means that our UDP audio isn't getting through.

Reply:
Yes, that sounds reasonable.